### PR TITLE
tests: remove centos 7 support

### PR DESCRIPTION
**Pull request by sergiocazzolato**
Centos-7 is EOL since June-30

Also included changes imported from snapd-testing-tools where also centos-7 support has been removed.

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -431,9 +427,6 @@ backends:
             - debian-sid-64:
                   username: debian
                   password: debian
-            - centos-7-64:
-                  username: centos
-                  password: centos
```
Comment (olivercalder) [inline spread.yaml:436]: Could this be replaced with centos 9?

File: tests/lib/external/snapd-testing-tools/spread.yaml [hunk 2]
```diff
@@ -47,7 +44,25 @@ backends:
...
+            NO_PROXY: 'canonical.com'
+        systems:
+            - ubuntu-20.04-64:
+                image: ubuntu-focal-20.04-amd64
+            - ubuntu-22.04-64:
+                image: ubuntu-jammy-22.04-amd64
+            - centos-9-64:
+            - debian-12-64:
+            - fedora-40-64:
+            - opensuse-15.5-64:
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/spread.yaml:65]: These changes seem unrelated, is this intentional?

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml [hunk 3]
```diff
@@ -143,12 +150,7 @@ execute: |
             os.query is-classic
             ! os.query is-core
             ;;
-        centos-7-64)            
-            os.query is-centos
-            os.query is-centos 7
             os.query is-classic
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:153]: Was this line supposed to be deleted as well?

Comment (olivercalder) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:153]: Or perhaps moved to below `centos-9-64)` ?

File: tests/lib/external/snapd-testing-tools/tests/remote.refresh/task.yaml [hunk 4]
```diff
@@ -15,14 +15,11 @@ environment:
...
     unit: nested-vm
-    TARGET/xenial: xenial
-    TARGET/bionic: bionic
+    TARGET: xenial
 
 prepare: |
     if [ "$TARGET" = xenial ]; then
         wget -O pc.img.xz https://storage.googleapis.com/snapd-spread-tests/images/booted/pc-amd64-16-stable-core_beta_2.56.2.img.xz
-    elif [ "$TARGET" = bionic ]; then
-        wget -O pc.img.xz https://storage.googleapis.com/snapd-spread-tests/images/booted/pc-amd64-18-stable-snapd_beta_2.56.2.img.xz
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/tests/remote.refresh/task.yaml:25]: More unrelated changes?

File: tests/lib/external/snapd-testing-tools/tests/remote.wait-for/task.yaml [hunk 5]
```diff
@@ -15,14 +15,11 @@ environment:
...
     unit: nested-vm
-    TARGET/xenial: xenial
-    TARGET/bionic: bionic
+    TARGET: xenial
 
 prepare: |
     if [ "$TARGET" = xenial ]; then
         wget -O pc.img.xz https://storage.googleapis.com/snapd-spread-tests/images/booted/pc-amd64-16-stable-core_beta_2.56.2.img.xz
-    elif [ "$TARGET" = bionic ]; then
-        wget -O pc.img.xz https://storage.googleapis.com/snapd-spread-tests/images/booted/pc-amd64-18-stable-snapd_beta_2.56.2.img.xz
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/tests/remote.wait-for/task.yaml:25]: Same

File: tests/lib/external/snapd-testing-tools/tools/tests.pkgs.pacman.sh [hunk 6]
```diff
@@ -25,7 +25,7 @@ remap_one() {
             echo "python-gobject"
             ;;
         test-snapd-pkg-1)
-            echo "freeglut"
+            echo "cmus"
```
Comment (olivercalder) [inline tests/lib/external/snapd-testing-tools/tools/tests.pkgs.pacman.sh:28]: Same

File: tests/main/interfaces-autopilot-introspection/task.yaml [hunk 7]
```diff
@@ -11,7 +11,6 @@ systems:
     - -ubuntu-14.04-*  # no tests.session
     - -ubuntu-core-*  # no session bus (except for core20+)
     - -amazon-linux-2-*  # no session bus
-    - -centos-7-*  # no session bus
```
Comment (olivercalder) [inline tests/main/interfaces-autopilot-introspection/task.yaml:14]: Does centos 9 support session bus, bfw module, python3 dbus bindings, quota groups, and the newer systemd so that all the below exclusions for centos 7 don't apply to centos 9?

**Review by olivercalder — COMMENTED** [review]
Thanks! A few questions about some seemingly-unrelated changes, and about where centos 7 should/not be replaced by centos 9.

File: spread.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply spread.yaml:436]: we are running in centos-9 already

File: tests/lib/external/snapd-testing-tools/spread.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/spread.yaml:65]: yes, because when I imported snapd-testing-tools to get the changes related to centos-7 I got the changes related to openstack as well that are already merged in snapd-testing-tools

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:153]: yes, it was moved to centos-9, this is how is it displayed on github

File: tests/lib/external/snapd-testing-tools/tests/remote.wait-for/task.yaml (hunk 5 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/remote.wait-for/task.yaml:25]: everything under tests/lib/external/snapd-testing-tools is the same

File: tests/lib/external/snapd-testing-tools/tools/tests.pkgs.pacman.sh (hunk 6 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tools/tests.pkgs.pacman.sh:28]: everything under tests/lib/external/snapd-testing-tools is the same

File: tests/main/interfaces-autopilot-introspection/task.yaml (hunk 7 above)
Comment (sergiocazzolato) [reply tests/main/interfaces-autopilot-introspection/task.yaml:14]: the test seems to be working in centos-9, just old versions had those session bus limitations

**Review by MiguelPires — APPROVED** [review]
thanks

File: spread.yaml (hunk 1 above)
Comment (olivercalder) [reply spread.yaml:436]: Not yet for the qemu backend, as far as I can tell.

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml (hunk 3 above)
Comment (olivercalder) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:153]: I just cloned the repo to check, and it looks like it is actually above the `centos-9` case:

```
        amazon-linux-2023-64)
            os.query is-amazon-linux
            os.query is-amazon-linux 2023
            ! os.query is-amazon-linux 2
            os.query is-classic
            ! os.query is-core
            ;;
            os.query is-classic
        centos-9-64)            
            os.query is-centos 9
            os.query is-centos
            ! os.query is-core
            ;;
        opensuse-15.5-64)
```

File: tests/lib/external/snapd-testing-tools/spread.yaml (hunk 2 above)
Comment (olivercalder) [reply tests/lib/external/snapd-testing-tools/spread.yaml:65]: Would this be fixed by a rebase? Or is there a separate repo for `snapd-testing-tools`?

**Review by olivercalder — COMMENTED** [review]
The `os.query is-classic` should be moved below `centos-9-64`, and I'd rather include `centos-9-64` for the qemu backend. Otherwise, looks good.

File: tests/lib/external/snapd-testing-tools/spread.yaml (hunk 2 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/spread.yaml:65]: we have a separate repo for snapd-testing-tools which is imported by using git subtree

File: tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml (hunk 3 above)
Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml:153]: I see something different:
https://github.com/canonical/snapd-testing-tools/blob/main/tests/os.query/task.yaml#L153

Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml]: the problem was caused by the merge tool, I just fixed that in snapd pr.

Comment (sergiocazzolato) [reply tests/lib/external/snapd-testing-tools/tests/os.query/task.yaml]: thanks for catching it

File: release/release_test.go [hunk 8]
```diff
@@ -142,20 +142,20 @@ BUG_REPORT_URL="https://bugs.launchpad.net/elementary/+filebug"`
...
 	mockOSRelease := filepath.Join(c.MkDir(), "mock-os-release")
 	dump := `NAME="CentOS Linux"
-VERSION="7 (Core)"
+VERSION="9 (Core)"
 ID="centos"
 ID_LIKE="rhel fedora"
-VERSION_ID="7"
-PRETTY_NAME="CentOS Linux 7 (Core)"
+VERSION_ID="9"
+PRETTY_NAME="CentOS Linux 9 (Core)"
```
[the remark this answers is not in the document]
Comment (sergiocazzolato) [reply release/release_test.go]: done

**Comment by sergiocazzolato** [comment]
@bboozzoo added change https://github.com/snapcore/snapd/pull/14157 to import more changed from snapd-testing-tools. Here I reverted most of the changes in that project

**Review by bboozzoo — APPROVED** [review]
LGTM
